tools/check/check-tidy.sh

```diff
@@ -22,4 +22,6 @@ set -euo pipefail
 # go mod tidy do not support symlink
 cd -P .
 
+cp go.sum /tmp/go.sum.before
 GO111MODULE=on go mod tidy
+diff -q go.sum /tmp/go.sum.before
```

Comment:
Could we use `git diff --quiet go.sum`?

Reply:
> Yes, with git diff there's no need to copy

If there're some uncommitted and valid changes in `go.sum`, `git diff` may report an unexpected error.

Reply:
Why would there be uncommitted changes in go.sum ?

Reply:
Developers don't always `git commit` before running `make check`

Reply:
oh, yes. I was thinking of the CI use case.